odbc/src/api/functions.rs

```diff
@@ -375,21 +409,57 @@ pub extern "C" fn SQLDriverConnect(
     _string_length_2: *mut SmallInt,
     _drive_completion: DriverConnectOption,
 ) -> SqlReturn {
+    dbg!();
     unsupported_function(MongoHandleRef::from(connection_handle), "SQLDriverConnect")
 }
 
 #[no_mangle]
 pub extern "C" fn SQLDriverConnectW(
-    _connection_handle: HDbc,
+    connection_handle: HDbc,
     _window_handle: HWnd,
-    _in_connection_string: *const WChar,
-    _string_length_1: SmallInt,
+    in_connection_string: *const WChar,
+    string_length_1: SmallInt,
     _out_connection_string: *mut WChar,
     _buffer_length: SmallInt,
     _string_length_2: *mut SmallInt,
     _driver_completion: DriverConnectOption,
 ) -> SqlReturn {
-    unimplemented!()
+    dbg!();
+    dbg!("!!!!");
+    let conn_handle = MongoHandleRef::from(connection_handle);
+    let conn = (*conn_handle).as_connection();
+    if conn.is_none() {
+        conn_handle.add_diag_info(ODBCError::InvalidHandleType("Connection"));
+        return SqlReturn::ERROR;
+    }
+    let conn = conn.unwrap();
+    let uri = input_wtext_to_string(in_connection_string, string_length_1 as usize);
+    let database = env::var("SQL_ATTR_CURRENT_CATALOG").ok();
```

Comment:
You should be able to retrieve the current catalog if set using ConnectionAttributes::current_db which should be present in the`attributes` struct of conn.
Now, because sqlSetConnAttr is not implemented, there is no way for a user to set it yet but `MongoConnection::connect` is an Option, so it should be alright.